ultradns/client.py

```diff
@@ -38,9 +41,14 @@ def _handle_error(self, response):
         if isinstance(json_body, list):
             error_code = json_body[0]['errorCode']
             error_msg = json_body[0]['errorMessage']
-        else:
+        elif isinstance(json_body, dict):
             error_code = json_body['errorCode']
             error_msg = json_body['errorMessage']
+        else:
+            error_code = ERR_HTTP
+            error_msg = (
+                'HTTP-level error. HTTP code: %s; response body: %s' % (
+                    response.status_code, response.content))
```

Comment:
This is already a separate case - so, you can just raise `HTTPError` directly here